packages/js/src/lib/init.ts

```diff
@@ -24,48 +33,101 @@ const addSessionEventListeners = (): void => {
   }
 };
 
-export const initialize = async (c: InitConfig): Promise<void> => {
+export const initialize = async (
+  c: InitConfig
+): Promise<Result<void, MissingSessionError | MissingFieldError | NetworkError | MissingPersonError>> => {
+  logger.debug("Start initialize");
+
   if (!c.environmentId) {
-    throw Error("Formbricks: environmentId is required");
+    logger.debug("No environmentId provided");
+    return err({
+      code: "missing_field",
+      field: "environmentId",
+    });
   }
+
   if (!c.apiHost) {
-    throw Error("Formbricks: apiHost is required");
+    logger.debug("No apiHost provided");
+
+    return err({
+      code: "missing_field",
+      field: "apiHost",
+    });
   }
+
   if (c.logLevel) {
+    logger.debug(`Setting log level to ${c.logLevel}`);
     logger.configure({ logLevel: c.logLevel });
   }
+
+  logger.debug("Adding widget container to DOM");
   addWidgetContainer();
+
+  logger.debug("Adding styles to DOM");
   addStylesToDom();
+
   if (!config || config.get().environmentId !== c.environmentId || config.get().apiHost !== c.apiHost) {
+    logger.debug("New config required");
     // we need new config
     config.update({ environmentId: c.environmentId, apiHost: c.apiHost });
-    // get person, session and settings from server
-    const { person, session, settings } = await createPerson();
+
+    logger.debug("Get person, session and settings from server");
+    const result = await createPerson();
+
+    if (result.ok !== true) {
+      return err(result.error);
+    }
+
+    const { person, session, settings } = result.value;
+
     config.update({ person, session: extendSession(session), settings });
     trackEvent("New Session");
   } else if (config.get().session && isExpired(config.get().session)) {
     // we need new session
-    const { session, settings } = await createSession();
+    const createSessionResult = await createSession();
+
+    if (createSessionResult.ok !== true) return err(createSessionResult.error);
+
+    const { session, settings } = createSessionResult.value;
+
     config.update({ session: extendSession(session), settings });
-    trackEvent("New Session");
-  } else if (!config.get().session) {
-    logger.error("Unable to initialize. No session found");
+
+    const trackEventResult = await trackEvent("New Session");
+
+    if (trackEventResult.ok !== true) return err(trackEventResult.error);
```

Comment:
I'm getting the error
`TypeError: Cannot read properties of undefined (reading 'ok')`